apkid/rules/elf/obfuscators.yara

```diff
@@ -130,7 +131,87 @@ rule ollvm_v6_0 : obfuscator
     $based_on      = "(based on Obfuscator-LLVM 6.0."
 
   condition:
-    all of them and not ollvm_v6_0_strenc
+    is_elf and
+    all of them and
+    not ollvm_v6_0_strenc
+}
+
+rule ollvm_v9_strenc : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM version 9.x (string encryption)"
+    sample      = "2314ec0053d829d424a82f702188fcb525cefce4feeef096f0855339b897a5d1"
+    author      = "Eduardo Novella"
+
+  strings:
+    $clang_version = /clang version \d\.\d\.\d /
+    $strenc        = /\.datadiv_decode[0-9]{18,20}/
+    $ollvm         = "(based on Obfuscator-LLVM 9."
+
+  condition:
+    is_elf and all of them
+}
+
+rule ollvm_tll : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM TLL (string encryption)"
+    url         = "https://github.com/yazhiwang/ollvm-tll.git"
+    sample      = "1f010330e9ac90f00d11aa37fdca25c437ad6f4b1302f6d7aa48b91ef22cc107"
+    author      = "Eduardo Novella"
+
+  strings:
+    /**
+      .datadiv_decode7760209850571766755
+      Android clang version 5.0.300080  (based on LLVM 5.0.300080)
+      clang version 6.0.0 (tags/RELEASE_600/final) (https://github.com/yazhiwang/ollvm-tll.git a38559e4c13359073102793c0a734bb1add3d5ff)
+    */
+    $clang_version = /clang version \d\.\d\.\d \(tags\/RELEASE\_\d{3}\/final\)/
+    $strenc        = /\.datadiv_decode[0-9]{18,20}/
+    $url           = "https://github.com/yazhiwang/ollvm-tll.git"
+
+  condition:
+    is_elf and all of them
+}
+
+rule ollvm_armariris : obfuscator
+{
+  meta:
+    description = "Armariris Obfuscator-LLVM (string encryption)"
+    url         = "https://github.com/GoSSIP-SJTU/Armariris"
+    sample      = "d22c2f53bab6fa2ab7bdb4f7acabb419e3ee3163bb758c4f7a013d07a8b09e12" // aka malware Joker
+    author      = "Eduardo Novella"
+
+  strings:
+    // clang version 3.9.1 (tags/RELEASE_391/final)
+    // clang version 5.0.1 (tags/RELEASE_501/final)
+    // .datadiv_decode14660921177804423408
+    $clang_version = /clang version \d\.\d\.\d \(tags\/RELEASE\_\d{3}\/final\)/
+    $strenc        = /\.datadiv_decode[0-9]{18,20}/
+
+  condition:
+    is_elf and
+    all of them and
+    not ollvm_tll and
+    for any i in (0..elf.symtab_entries): (elf.symtab[i].name contains ".datadiv_decode")
+}
+
+rule ollvm_strenc : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM alike (string encryption)"
```

Comment:
What does "alike" mean in this context? If it means unknown version, then let's use that to make it consistent with the other rules that do this.

Reply:
Replace `s/alike/unknown/g`. It is important to keep the string encryption in the title of the rule and avoid merging both rules in one.

Reply:
I'm still not sure what "alike" means here. I'm assuming it means unknown version. I'm not suggesting removing "string encryption", just changing the word "alike" to "unknown version". E.g. `Obfuscator-LLVM (unknown version, string encryption)`. This will make it more consistent with similar rules.

If it doesn't really mean unknown version and means "for some reason we didn't detect other stuff we expected to see", I think "unknown version" still makes sense. Thoughts?